ethdb/table.go

```diff
@@ -296,7 +291,7 @@ func (t *Table) Put(key, value []byte) error {
 		return nil
 	}
 
-	s, err := t.CreateSegmentIfNotExists(t.Partitioner.Partition(key))
+	s, err := t.CreateSegmentIfNotExists(context.TODO(), t.Partitioner.Partition(key))
```

Comment:
Still not linked up to the other end but any higher than this gets in to the core DB interfaces, so too much for this PR.